### PR TITLE
Transport Integration - Space Notify Send & Receive

### PR DESCRIPTION
**Pull request by neonphog**
- Cleans up some of the apis & adds some events to the top-level handler
- Kitsune top-module creates a tx instance and passes it to spaces
- Space module registers space handlers with the tx module
- Protocol and implementation of space send & recv

### CODE REVIEW

File: crates/api/src/kitsune.rs [hunk 1]
```diff
@@ -5,6 +5,18 @@ use std::sync::Arc;
 
 /// Handler for events coming out of Kitsune2.
 pub trait KitsuneHandler: 'static + Send + Sync + std::fmt::Debug {
+    /// A notification that a new listening address has been bound.
+    /// Peers should now go to this new address to reach this node.
+    fn new_listening_address(&self, this_url: Url) {
+        drop(this_url);
```
Comment (jost-s) [inline crates/api/src/kitsune.rs:11]: What are these `drop`s for?

**Review by jost-s — APPROVED** [review]

File: crates/api/src/kitsune.rs (hunk 1 above)
Comment (neonphog) [reply crates/api/src/kitsune.rs:11]: Underscores at the beginning of parameters show up in the docs, so I don't like them there. The explicit drops prevent the unused parameter warning. These default implementations don't need to do anything.

Comment (jost-s) [reply crates/api/src/kitsune.rs:11]: But you want a default implementation that does nothing, I guess. Alright, got it.

**Review by jost-s — APPROVED** [review]

**Review by ThetaSinner — APPROVED** [review]
LGTM!

**Pull request merged**